modules/extension/transform/src/main/java/org/geotools/data/transform/TransformDataStoreFactory.java

```diff
@@ -0,0 +1,181 @@
+    public static final Param REPOSITORY =
+            new Param(
+                    "Repository",
+                    Repository.class,
+                    "Class name for data store repository implementation",
+                    true);
+
+    public static final Param DATASTORE_NAME =
+            new Param("DataStoreName", String.class, "DataStore Name", true);
+
+    public static final Param FEATURE_TYPE_NAME =
+            new Param("FeatureTypeName", String.class, "FeatureTypeName / Layer Name", true);
+
+    public static final Param NEW_FEATURE_TYPE_NAME =
+            new Param("NewFeatureTypeName", String.class, "New FeatureTypeName", true);
+
+    public static final Param TRANSFORM_DEFINITION =
+            new Param(
+                    "TransformDefinition",
+                    String.class,
+                    "Transform definition for columns to add",
+                    false);
+
+    public static final Param ATTRIBUTE_SUBSET =
+            new Param(
+                    "AttributeSubset",
+                    String.class,
+                    "Comma-separated subset of columns to retain.  Leave empty to keep all columns.",
+                    false);
+
+    public static final Param NAMESPACEP =
+            new Param("namespace", URI.class, "uri to a the namespace", false); // not required
+
+    @Override
+    public DataStore createDataStore(Map<String, Serializable> params) throws IOException {
+
+        URI namespace = (URI) NAMESPACEP.lookUp(params);
+        Repository repository = (Repository) REPOSITORY.lookUp(params);
+
+        String datastoreName = (String) DATASTORE_NAME.lookUp(params);
+        String featureTypeName = (String) FEATURE_TYPE_NAME.lookUp(params);
+        String newFeatureTypeName = (String) NEW_FEATURE_TYPE_NAME.lookUp(params);
+
+        DataStore ds = repository.dataStore(new NameImpl(datastoreName));
+        SimpleFeatureSource inputFS = ds.getFeatureSource(featureTypeName);
+
+        String attributeSubset = (String) ATTRIBUTE_SUBSET.lookUp(params);
+
+        List<Definition> definitions = null;
+
+        if (attributeSubset != null) {
+
+            definitions =
+                    Arrays.stream(attributeSubset.split(","))
+                            .map(Definition::new)
+                            .collect(Collectors.toCollection(ArrayList::new));
+        } else {
+            definitions =
+                    inputFS.getSchema()
+                            .getAttributeDescriptors()
+                            .stream()
+                            .map(ad -> new Definition(ad.getLocalName()))
+                            .collect(Collectors.toList());
+        }
+
+        String transformDefinition = (String) TRANSFORM_DEFINITION.lookUp(params);
+
+        if (transformDefinition != null) {
+            List<Definition> transforms =
+                    Arrays.stream(transformDefinition.split(";"))
```

Comment:
How about using a property file format instead? The GUI in GeoServer allows for textarea input, if the Param is marked as "large text", example here: https://github.com/geotools/geotools/blob/14d27a2db448454c64e8f38e9977565cc400d52b/modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStoreFactory.java#L218